PGPsendmail-v1/pgpsendmail/man/man1/extract-addrs.1
.\" Copyright (C) 1997  Richard Gooch
.\"
.\" This program is free software; you can redistribute it and/or modify
.\" it under the terms of the GNU General Public License as published by
.\" the Free Software Foundation; either version 2 of the License, or
.\" (at your option) any later version.

.\" This program is distributed in the hope that it will be useful,
.\" but WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\" GNU General Public License for more details.
.\"
.\" You should have received a copy of the GNU General Public License
.\" along with this program; if not, write to the Free Software
.\" Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
.\"
.\" Richard Gooch may be reached by email at  rgooch@atnf.csiro.au
.\" The postal address is:
.\"   Richard Gooch, c/o ATNF, P. O. Box 76, Epping, N.S.W., 2121, Australia.
.\"
.\"	extract-addrs.1		Richard Gooch	27-SEP-1997
.\"
.TH EXTRACT-ADDRS 1 "27 Sep 1997" "PGPsendmail/Unix Suite"
.SH NAME
extract-addrs \- extract email addresses
.SH SYNOPSIS
.B extract-addrs
[
.B t
]
[
.B c
]
[
.B b
]
[
.B f
]
[
.B s
]
.SH DESCRIPTION
The \fIextract-addrs\fP programme will extract email addresses from an
email header. The header is read from the standard input. Each email
address extracted is written to the standard output, each in a
separate line. This programme can be used as the basis for an advanced
email-address matching system for mail filters.
.SH OPTIONS
.TP
.B t
Extract email addresses from the "To:" field.
.TP
.B c
Extract email addresses from the "Cc:" field.
.TP
.B b
Extract email addresses from the "Bcc:" field.
.TP
.B f
Extract email addresses from the "From:" and "From\ " fields.
.TP
.B s
Extract email addresses from the "Sender:" field.
.SH DEFAULT BEHAVIOUR
If no options are given, the default "fs" options are used.
.SH SEE ALSO
.BR pgpsendmail (1),
.BR match-email-addr (1)
.SH AUTHOR
Richard Gooch (rgooch@atnf.csiro.au)
.SH AVAILABILITY
The PGPsendmail/Unix Suite is available for anonymous ftp from:

ftp.atnf.csiro.au:pub/people/rgooch
ftp.dhp.com:pub/crypto/pgp/PGPsendmail
ftp.ox.ac.uk:pub/crypto/pgp/utils
linux.nrao.edu:pub/src/pgp/PGPsendmail


rrc2/history
 ############################################################################
#                                                                            #
#                      History of my Implementation of                       #
#                      RRC.2 - Ron Rivest Cipher Nr. 2                       #
#                                                                            #
 ############################################################################

13.02.96:
	Received specification

15.02.96: 
	Linux (ANSI-) C version done;
	includes RRC.2 library, test program, and simple sample crypt
	program

18.02.96:
	First announcement on usenet (rrc2_news.txt)

18.02.96:
	Turbo Pascal version done;
	Includes only Unit RRC2 and test program.

20.02.96:
	Improved C version: now endian neutral (untested)

20.02.96:
	RRC.2 obtainable via ftp

23.02.96:
	second usenet announcement (rrc2_news_2.txt)

# version FINI


PGPsendmail-v1/pgpsendmail/extract-addrs.c
/*  extract-addrs.c

    Source file for extracting email addresses from a header.

    Copyright (C) 1997  Richard Gooch

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.

    Richard Gooch may be reached by email at  rgooch@atnf.csiro.au
    The postal address is:
      Richard Gooch, c/o ATNF, P. O. Box 76, Epping, N.S.W., 2121, Australia.
*/

/*  This file is a utility programme to extract email addresses from a mail
    header.


    Written by      Richard Gooch   27-SEP-1997

    Last updated by Richard Gooch   27-SEP-1997


*/
#include <stdio.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <sys/wait.h>
#include <signal.h>
#include <ctype.h>
#include <unistd.h>
#include <stdlib.h>
#include <fcntl.h>
#include <string.h>
#include <errno.h>
#include "pgpsendmail.h"


/*  Private functions  */


/*  Public functions follow  */

int main (int argc, char **argv)
{
    flag eof;
    int count;
    decode_type info;
    char *ptr;
    char error[STRING_LENGTH];
    static char usage_string[] = "Usage:\textract-addrs [tcbfs]";

    m_clear ( (char *) &info, sizeof info );
    if (argc == 1)
    {
	info.scan_from = TRUE;
	info.scan_sender = TRUE;
    }
    for (--argc, ++argv; argc > 0; --argc, ++argv)
    {
	for (ptr = *argv; *ptr != '\0'; ++ptr)
	{
	    switch (*ptr)
	    {
	      case 't':
	      case 'T':
		info.scan_to = TRUE;
		break;
	      case 'c':
	      case 'C':
		info.scan_cc = TRUE;
		break;
	      case 'b':
	      case 'B':
		info.scan_bcc = TRUE;
		break;
	      case 'f':
	      case 'F':
		info.scan_from = TRUE;
		break;
	      case 's':
	      case 'S':
		info.scan_sender = TRUE;
		break;
	      default:
		fprintf (stderr, "Illegal option: %c\n", *ptr);
		exit (1);
		/*break;*/
	    }
	}
    }
    while ( read_line (INPUT_FD, &info, NULL, &eof) )
    {
	if ( eof || (info.line_length < 1) )
	{
	    fflush (stdout);
	    exit (0);
	}
	while ( ( ptr = get_next_address (&info, error) ) != NULL )
	{
	    puts (ptr);
	}
	if (error[0] == '\0') continue;
	fprintf (stderr, "Error decoding, error: \"%s\"\n", error);
	exit (1);
    }
    fprintf (stderr, "Error reading\n");
    exit (1);
}   /*  End Function main  */
